test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -231,25 +241,37 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
 		}
 	}
 
-	// Collect all secondary network IPs when they are available.
+	// Collect all secondary network IPs and verify MACs when they are available.
 	for _, testPod := range data.pods {
 		_, err := e2eTestData.PodWaitFor(defaultTimeout, testPod.podName, namespace, func(pod *corev1.Pod) (bool, error) {
 			if pod.Status.Phase != corev1.PodRunning {
 				return false, nil
 			}
 			var podNetworkAttachments []*attachment
-			podIPs, _, _, err := data.listPodAddresses(testPod)
+			podIPs, _, macResult, err := data.listPodAddresses(testPod)
 			if err != nil {
 				return false, err
 			}
 			for iface, net := range testPod.interfaceNetworks {
-				if podIPs[iface] == nil {
+				ip := podIPs[iface]
+				if ip == nil {
 					return false, nil
 				}
+
+				// Verify MAC address
```

Comment:
I'd personally rather do this in a separate function. If kept here, then the function comment for `pingBetweenInterfaces` should be updated, and maybe the function should have a more generic name?

Reply:
done, i updated existing function.